{
    "status": "Fix Released", 
    "last_updated": "2013-04-04 11:19:41.414140+00:00", 
    "description": "The pep8 for patch set 6 of https://review.openstack.org/#/c/22707 returned invalid results. http://logs.openstack.org/22707/6/check/gate-nova-pep8/21479/console.html.gz\n\n\npep8 reported \n'./nova/virt/vmwareapi/vim.py:32:1: N302  import only modules.'from nova.virt.vmwareapi import error_util' does not import a module'\n\n because https://review.openstack.org/#/c/22707/6..7/nova/openstack/common/db/exception.py\n had in invalid import.\n\nThe current logic compares the last word of the module name along with the last word of the exception msg, and if they are not the same a module is missing.  This check should also make sure the words \"cannot import\" aren't in the exception msg.", 
    "tags": [], 
    "importance": "Undecided", 
    "heat": 6, 
    "link": "https://bugs.launchpad.net/nova/+bug/1133103", 
    "owner": "https://api.launchpad.net/1.0/~jogo", 
    "id": 1133103, 
    "index": 3444, 
    "created": "2013-02-26 03:40:49.904960+00:00", 
    "title": "pep8 returns false N302 for missing import", 
    "comments": [
        {
            "content": "The pep8 for patch set 6 of https://review.openstack.org/#/c/22707 returned invalid results. http://logs.openstack.org/22707/6/check/gate-nova-pep8/21479/console.html.gz\n\n\npep8 reported \n'./nova/virt/vmwareapi/vim.py:32:1: N302  import only modules.'from nova.virt.vmwareapi import error_util' does not import a module'\n\n because https://review.openstack.org/#/c/22707/6..7/nova/openstack/common/db/exception.py\n had in invalid import.\n\nThe current logic compares the last word of the module name along with the last word of the exception msg, and if they are not the same a module is missing.  This check should also make sure the words \"cannot import\" aren't in the exception msg.", 
            "date_created": "2013-02-26 03:40:49.904960+00:00", 
            "author": "https://api.launchpad.net/1.0/~jogo"
        }, 
        {
            "content": "Fix proposed to branch: master\nReview: https://review.openstack.org/22947", 
            "date_created": "2013-02-26 03:48:27.982890+00:00", 
            "author": "https://api.launchpad.net/1.0/~hudson-openstack"
        }, 
        {
            "content": "Reviewed:  https://review.openstack.org/22947\nCommitted: http://github.com/openstack/nova/commit/99308d4dd3b16e00af4e610be4f07c7d99e49846\nSubmitter: Jenkins\nBranch:    master\n\ncommit 99308d4dd3b16e00af4e610be4f07c7d99e49846\nAuthor: Joe Gordon <email address hidden>\nDate:   Mon Feb 25 19:44:29 2013 -0800\n\n    Fix hacking.py to handle 'cannot import x'\n    \n    Fix missingImport logic to handle a module that cannot be imported, due\n    to a missing dependant module.\n    \n    Due to the nature of this bug doctests cannot cover it.\n    \n    Fix bug 1133103\n    \n    Change-Id: Ia067bacc125b67e2d9e0cfc72495e4f13a8648ad\n", 
            "date_created": "2013-02-26 06:07:14.126151+00:00", 
            "author": "https://api.launchpad.net/1.0/~hudson-openstack"
        }
    ]
}